src/app/core/services/show-style-variant-state.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable, OnDestroy } from "@angular/core";
+import { BehaviorSubject, lastValueFrom, Subscription, SubscriptionLike } from "rxjs";
+import { ShowStyleVariant } from "../models/show-style-variant";
+import { ShowStyleVariantService } from "../abstractions/show-style-variant.service";
+import { EventSubscription } from "../../event-system/abstractions/event-observer.service";
+import { ConnectionStatusObserver } from "./connection-status-observer.service";
+import { ManagedSubscription } from "./managed-subscription.service";
+
+
+@Injectable()
+export class ShowStyleVariantStateService implements OnDestroy {
+    private readonly showStyleVariantSubjects: Map<string, BehaviorSubject<ShowStyleVariant>> = new Map()
+    private eventSubscriptions: EventSubscription[]
+
+    constructor(
+        private readonly showStyleVariantService: ShowStyleVariantService,
+        private readonly connectionStatusObserver: ConnectionStatusObserver
+    ) {
+        this.subscribeToEvents()
+    }
+
+    private subscribeToEvents(): void {
+        this.eventSubscriptions = [
+            ...this.subscribeToShowStyleVariantEvents(),
+            ...this.subscribeToConnectionStatus(),
+        ]
+    }
+
+    private subscribeToConnectionStatus(): EventSubscription[] {
+        return [
+            this.connectionStatusObserver.subscribeToReconnect(this.resetShowStyleVariantSubjects.bind(this))
+        ]
+    }
+
+    private subscribeToShowStyleVariantEvents(): EventSubscription[] {
+        return []
+    }
+
+    private resetShowStyleVariantSubjects(): void {}
```

Comment:
Is this suppose to be an empty private method?

Reply:
Actually yes, for now it is there to follow the structure of something having to happen on a reconnect event. I don't appreciate this boiler code though.

Reply:
Sound like we have the structure elsewhere, which means we can look it up, how it is suppose to be when needed. 
That also means that for now, you should be able to remove it without issue.

Reply:
A very good point, i decided to remove most of the structure!